### PR TITLE
Update Node version to 8 and 10

### LINKED ISSUE #439: Travis Failing for all of the latest PRs
I see most of the recent PRs are all failing. Any help is highly appreciated.

Error
```
$ grunt build
/home/travis/build/publiclab/PublicLab.Editor/node_modules/grunt/node_modules/mkdirp/index.js:4
const {mkdirpNative, mkdirpNativeSync} = require('./lib/mkdirp-native.js')
```

```
0.29s$ grunt build
/home/travis/build/publiclab/PublicLab.Editor/node_modules/grunt/node_modules/mkdirp/lib/opts-arg.js:7
    opts = { mode: 0o777 & (~process.umask()), fs, ...opts }
                                                   ^^^
```

```
 grunt build
/home/travis/build/publiclab/PublicLab.Editor/node_modules/grunt/node_modules/mkdirp/lib/opts-arg.js:7
    opts = { mode: 0o777 & (~process.umask()), fs, ...opts }
                                                   ^^^
SyntaxError: Unexpected token ...
  at createScript (vm.js:53:10)
  at Object.runInThisContext (vm.js:95:10)
  at Module._compile (module.js:543:28)
  at Object.Module._extensions..js (module.js:580:10)
  at Module.load (/home/travis/build/publiclab/PublicLab.Editor/node_modules/coffeescript/lib/coffee-script/register.js:45:36)
  at tryModuleLoad (module.js:447:12)
  at Function.Module._load (module.js:439:3)
  at Module.require (module.js:498:17)
  at require (internal/module.js:20:19)
  at Object.<anonymous> (/home/travis/build/publiclab/PublicLab.Editor/node_modules/grunt/node_modules/mkdirp/index.js:1:79)
  at Module._compile (module.js:571:32)
  at Object.Module._extensions..js (module.js:580:10)
  at Module.load (/home/travis/build/publiclab/PublicLab.Editor/node_modules/coffeescript/lib/coffee-script/register.js:45:36)
  at tryModuleLoad (module.js:447:12)
  at Function.Module._load (module.js:439:3)
  at Module.require (module.js:498:17)
  at require (internal/module.js:20:19)
  at Object.<anonymous> (/home/travis/build/publiclab/PublicLab.Editor/node_modules/grunt/lib/grunt/file.js:20:14)
  at Module._compile (module.js:571:32)
  at Object.Module._extensions..js (module.js:580:10)
  at Module.load (/home/travis/build/publiclab/PublicLab.Editor/node_modules/coffeescript/lib/coffee-script/register.js:45:36)
  at tryModuleLoad (module.js:447:12)
  at Function.Module._load (module.js:439:3)
  at Module.require (module.js:498:17)
  at require (internal/module.js:20:19)
  at gRequire (/home/travis/build/publiclab/PublicLab.Editor/node_modules/grunt/lib/grunt.js:14:24)
  at Object.<anonymous> (/home/travis/build/publiclab/PublicLab.Editor/node_modules/grunt/lib/grunt.js:28:1)
  at Module._compile (module.js:571:32)
  at Object.Module._extensions..js (module.js:580:10)
  at Module.load (module.js:488:32)
  at tryModuleLoad (module.js:447:12)
  at Function.Module._load (module.js:439:3)
  at Module.require (module.js:498:17)
  at require (internal/module.js:20:19)
  at Object.<anonymous> (/home/travis/build/publiclab/PublicLab.Editor/node_modules/grunt/node_modules/grunt-cli/bin/grunt:44:1)
  at Module._compile (module.js:571:32)
  at Object.Module._extensions..js (module.js:580:10)
  at Module.load (module.js:488:32)
  at tryModuleLoad (module.js:447:12)
  at Function.Module._load (module.js:439:3)
  at Module.require (module.js:498:17)
  at require (internal/module.js:20:19)
  at Object.<anonymous> (/home/travis/build/publiclab/PublicLab.Editor/node_modules/grunt/bin/grunt:3:1)
  at Module._compile (module.js:571:32)
  at Object.Module._extensions..js (module.js:580:10)
  at Module.load (module.js:488:32)
  at tryModuleLoad (module.js:447:12)
  at Function.Module._load (module.js:439:3)
  at Module.runMain (module.js:605:10)
  at run (bootstrap_node.js:427:7)
  at startup (bootstrap_node.js:151:9)
  at bootstrap_node.js:542:3
```

### PR DESCRIPTION
**Pull request by keshav234156**
Fixes #439 #205

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `grunt jasmine`
* [ ] code is in uniquely-named feature branch and has no merge conflicts
* [ ] PR is descriptively titled
* [ ] PR body includes `fixes #0000`-style reference to original issue #
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. 

Thanks!

### CODE REVIEW

**Comment by keshav234156** [comment]
@cesswairimu

**Comment by cesswairimu** [comment]
@jywarren @emilyashley could you please merge this one...Got no merging rights on this repo

**Comment by keshav234156** [comment]
@HarshKhandeparkar @VladimirMikulic strangely tests are working fine on Node 8 but are failing on Node 10. I think we also have faced a similar issue before https://github.com/publiclab/PublicLab.Editor/issues/205. Any ideas about this

**Comment by HarshKhandeparkar** [comment]
Can you post what the error is, here?

**Comment by VladimirMikulic** [comment]
@keshav234156 since Node v10, `fs.writeFile` requires a callback function to be passed in. One of the dependencies is not passing the callback at throws an error.

**Comment by keshav234156** [comment]
```
Fatal error: Callback must be a function
The command "grunt jasmine" exited with 3.
```

**Comment by VladimirMikulic** [comment]
You would need to dig into `node_modules` in order to fix this. It's a problem with Jasmine.
Read more [here](https://github.com/gruntjs/grunt-contrib-jasmine/issues/266).

**Comment by HarshKhandeparkar** [comment]
Is jasmine version up to date?

**Comment by keshav234156** [comment]
```
Fatal error: Callback must be a function
``` 
is also present in Node 8 and error was also thrown in earlier versions of Node but it didn't failed the travis.

In Node 10 the extra error is 
```
The command "grunt jasmine" exited with 3.
```

**Comment by VladimirMikulic** [comment]
I guess our setup is outdated.

**Comment by HarshKhandeparkar** [comment]
`grunt jasmine` exited with 3 is not an extra error. It just tells you that
nodejs v10 handles the errors in a different way.

When the same error occurred in v8, nodejs caught the error but didn't stop
the script or return a non-zero exit code.

In v10, it caught the error, stopped the execution and reported to travis
that the script had failed.

**Comment by keshav234156** [comment]
@VladimirMikulic @HarshKhandeparkar Fixed by updataing grunt-jasmine-contrib

**Comment by HarshKhandeparkar** [comment]
Great! Always update dependencies!

File: package.json [hunk 1]
```diff
@@ -44,15 +44,15 @@
   },
   "devDependencies": {
     "browserify": "16.5.0",
-    "grunt": "^1.0.0",
+    "grunt": "^1.0.4",
```
Comment (alaxalves) [inline package.json]: I don't think it's a good idea using the '^'. We'll always use the edge version, and the behavior of the grunt lib might change.

Comment (HarshKhandeparkar) [reply package.json]: We use it with `^` in ImageSequencer. That is not a problem. It is advisable to always use the latest version.

Comment (VladimirMikulic) [reply package.json]: Correct, `^` won't update the major version so as @HarshKhandeparkar it is advisable to use it.

**Comment by jywarren** [comment]
Looks great folks, thanks a ton!

**Pull request merged**